std/experimental/allocator/common.d

```diff
@@ -287,16 +287,19 @@ Returns the effective alignment of `ptr`, i.e. the largest power of two that is
 a divisor of `ptr`.
 */
 @nogc nothrow pure
-package uint effectiveAlignment(void* ptr)
+package size_t effectiveAlignment(void* ptr)
 {
-    return 1U << trailingZeros(cast(size_t) ptr);
+    return (cast(size_t) 1) << trailingZeros(cast(size_t) ptr);
```

Comment:
Isn't it possible for trailingZeros to return `64`?

```
assert(trailingZeros(0) == 64);
```

So maybe `trailingZeros` needs to be fixed to accept `size_t` too?

Reply:
Right, good point, so for 32-bit `size_t(0)`, trailingZeros() would indeed return 64 as well. Here, it doesn't matter, as `effectiveAlignment(null)` returns `size_t(1) << 64U => 1` for both 32/64-bit size_t.

Reply:
Style nit: prefer `size_t(1)`